symbolic-debuginfo/src/wasm.rs

```diff
@@ -292,67 +258,39 @@ impl<'data: 'object, 'object> ObjectLike<'data, 'object> for WasmObject<'data> {
     }
 }
 
-impl<'d> Dwarf<'d> for WasmObject<'d> {
+impl<'data> Dwarf<'data> for WasmObject<'data> {
     fn endianity(&self) -> Endian {
         Endian::Little
     }
 
-    fn raw_section(&self, section_name: &str) -> Option<DwarfSection<'d>> {
-        for (_, section) in self.wasm_module.customs.iter() {
-            if section.name().strip_prefix('.') == Some(section_name) {
-                return Some(DwarfSection {
-                    data: Cow::Owned(section.data(&Default::default()).into_owned()),
+    fn raw_section(&self, section_name: &str) -> Option<DwarfSection<'data>> {
+        self.dwarf_sections.iter().find_map(|(name, data)| {
+            if name.strip_prefix('.') == Some(section_name) {
+                Some(DwarfSection {
+                    data: Cow::Borrowed(data),
                     // XXX: what are these going to be?
                     address: 0,
                     offset: 0,
                     align: 4,
-                });
+                })
+            } else {
+                None
             }
-        }
-
-        None
+        })
     }
 }
 
 /// An iterator over symbols in the WASM file.
 ///
 /// Returned by [`WasmObject::symbols`](struct.WasmObject.html#method.symbols).
-pub struct WasmSymbolIterator<'data, 'object> {
-    funcs: std::iter::Peekable<Box<dyn Iterator<Item = &'object walrus::Function> + 'object>>,
-    _marker: std::marker::PhantomData<&'data [u8]>,
+pub struct WasmSymbolIterator<'data> {
+    funcs: Vec<Symbol<'data>>,
```

Comment:
or maybe use a `std::vec::IntoIter` so you can avoid the `reverse()`.

Also, not quite sure if removing the lifelite parameter is considered a breaking change? If so, maybe add a `PhantomData`.

Reply:
Not sure what you mean, how would into_iter() avoid the `reverse()`?

Reply:
I think `IntoIter` yields from the front without needing to move data around.

Reply:
Sorry I was uncaffeinated at the time, I've preserved the original API now.